### PR TITLE
Improve SolutionTransfer documentation

### PR DESCRIPTION
**Pull request by jppelteret**
The documentation for the SolutionTransfer class now includes a note on how to use with MPI partitioned grids. I've added a couple of tests that I used to teach myself how this is supposed to work.

### CODE REVIEW

File: tests/mpi/solution_transfer_02.cc [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  DoFHandler<dim> dof_handler(tria);
+
+  TrilinosWrappers::MPI::Vector solution;
+
+  dof_handler.distribute_dofs(fe);
+  IndexSet locally_owned_dofs =
+    DoFTools::locally_owned_dofs_per_subdomain(dof_handler)[this_mpi_process];
+  IndexSet locally_relevant_dofs =
+    DoFTools::locally_relevant_dofs_per_subdomain(
+      dof_handler)[this_mpi_process];
```
Comment (drwells) [inline tests/mpi/solution_transfer_02.cc]: same

File: tests/hp/solution_transfer_14.cc [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  dof_handler.begin(0)->child(0)->set_active_fe_index(1);
+
+  TrilinosWrappers::MPI::Vector solution;
+
+  dof_handler.distribute_dofs(fe);
+  IndexSet locally_owned_dofs =
+    DoFTools::locally_owned_dofs_per_subdomain(dof_handler)[this_mpi_process];
+  IndexSet locally_relevant_dofs =
+    DoFTools::locally_relevant_dofs_per_subdomain(
+      dof_handler)[this_mpi_process];
```
[the remark this answers is not in the document]
Comment (jppelteret) [reply tests/hp/solution_transfer_14.cc]: I can make the change in the documentation, but not the tests. The tests use a standard `Triangulation` partitioned with `metis` (hence my use of the "old school" method of DoF indexset extraction. With your proposed changes the vector `reinit` call fails the condition `!has_ghost_elements()`.

File: include/deal.II/numerics/solution_transfer.h [hunk 3]
```diff
@@ -113,6 +113,36 @@ DEAL_II_NAMESPACE_OPEN
  * soltrans.interpolate(solution, interpolated_solution);
  * @endcode
  *
+ * <li> If the grid is distributed, or partitioned across several MPI processes,
+ * then it is important to note that the old solution(s) must be copied to one
+ * that also provides access to the locally relevant DoF values (these values
+ * required for the interpolation process):
+ * @code
+ * IndexSet locally_owned_dofs = DoFTools::locally_owned_dofs_per_subdomain(
+ *   dof_handler)[this_mpi_process];
```
[the remark this answers is not in the document]
Comment (bangerth) [reply include/deal.II/numerics/solution_transfer.h]: It doesn't matter -- the array you're accessing here is communicated once at the end of `distribute_dofs()`, not each time the function above is called.

File: include/deal.II/numerics/solution_transfer.h [hunk 4]
```diff
@@ -113,6 +113,35 @@ DEAL_II_NAMESPACE_OPEN
  * soltrans.interpolate(solution, interpolated_solution);
  * @endcode
  *
+ * <li> If the grid is distributed, or partitioned across several MPI processes,
```
Comment (bangerth) [inline include/deal.II/numerics/solution_transfer.h]: Should this discussion not be part of the `parallel::distributed::SolutionTransfer` class?

Comment (jppelteret) [reply include/deal.II/numerics/solution_transfer.h]: As you can see from the tests its relevant for the standard `SolutionTransfer` class when using `Triangulation` + `Metis`/`Zoltan` as partitioners. So I think that it should stay here, but maybe with a clarification of that point. Would you like me to copy it over to the `p::d::ST` documentation as well?

Comment (bangerth) [reply include/deal.II/numerics/solution_transfer.h]: Yes, maybe. Thanks!

Is there any mention of `p::d::ST` anywhere here at all?

Comment (jppelteret) [reply include/deal.II/numerics/solution_transfer.h]: @bangerth Sorry its taken me so long to get to this. I've now added a blurb in the introduction to `p::d::ST`.

**Comment by masterleinad** [comment]
@drwells ping

**Comment by drwells** [comment]
Looks good!

File: include/deal.II/numerics/solution_transfer.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (drwells) [reply include/deal.II/numerics/solution_transfer.h]: I must admit that I had never read the detailed description of this function: you're right, of course (there is no communication).

**Pull request merged**